verify/main_test.go

```diff
@@ -2,72 +2,98 @@ package main
 
 import (
 	"context"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"net/http/httptest"
 	"os"
 	"testing"
+	"time"
 
 	"cloud.google.com/go/firestore"
 	"github.com/aws/aws-lambda-go/events"
 	"github.com/stretchr/testify/assert"
-
-	"fmt"
-	"google.golang.org/api/iterator"
 )
 
-func TestHandler(t *testing.T) {
+func newFirestoreMockClient(ctx context.Context) *firestore.Client {
+	client, err := firestore.NewClient(ctx, "test")
+	if err != nil {
+		log.Fatalf("firebase.NewClient err: %v", err)
+	}
+
+	return client
+}
 
+func TestHandler(t *testing.T) {
 	os.Setenv("environment", "test")
 	defer os.Unsetenv("environment")
 
-	var RefId string = ""
+	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
+	client := newFirestoreMockClient(ctx)
+	defer cancel()
 
-	ctx := context.Background()
-	client, err := firestore.NewClient(ctx, "test")
-	if err != nil {
-		return
-	}
+	server1 := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+	}))
+	defer server1.Close()
 
-	client.Collection("users").Add(ctx, map[string]interface{}{
-		"chaincode":  "abcdefgh",
-		"profileURL": "https://96phoonyw3.execute-api.us-east-2.amazonaws.com/Prod",
-	})
+	server2 := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+		w.WriteHeader(http.StatusOK)
+		w.Write([]byte(`{"hash": "$2a$12$ScGc2Q0t0rqqSJK1E2W/WuaRVAchaVWdUqb1hQi21cFTnOVvlIdry"}`))
+	}))
+	defer server2.Close()
```

Comment:
here you are creating a custom server, not using the one which is used in the actual code/file. You can make the server created in verify file exportable and import it here to send a HTTP request.

Reply:
Can you give me some idea on what you are referring to when you say the server created in verify file?